test/subspace-test-service/Cargo.toml

```diff
@@ -30,7 +30,7 @@ sp-blockchain = { git = "https://github.com/subspace/substrate", rev = "b49d8ee5
 sp-keyring = { git = "https://github.com/subspace/substrate", rev = "b49d8ee54f4bd6e4b5b48f45d08f75705fad95b4" }
 sp-runtime = { git = "https://github.com/subspace/substrate", rev = "b49d8ee54f4bd6e4b5b48f45d08f75705fad95b4" }
 subspace-runtime-primitives = { path = "../../crates/subspace-runtime-primitives" }
-subspace-service = { path = "../../crates/subspace-service" }
+subspace-service = { path = "../../crates/subspace-service", features = ["testing"] }
```

Comment:
It should be an explicit library parameters. By doing it this way you cause feature unification and nothing else would be able to test `subspace-service` with `testing` feature disabled if they test the whole repo at once.

Reply:
Make sense, although unused at present. Made it a default feature as this is a test-specific library.

Reply:
I don't really see anything changed with latest commit. Feature unification still takes place. We can't use features for this kind of setting here at all. If you build tests and binaries at the same time, your binaries will have `testing` feature enabled and we certainly don't want that.

Reply:
You're right, I didn't have a good understanding of feature unification of cargo, but now I learned a bit (also discovered [an issue you created](https://github.com/rust-lang/cargo/issues/9949) before :P). My idea at present is to probably add a new argument to `new_full` to avoid the hassle of using a feature for such a thing, what do you think?